include/aikido/constraint/detail/JointStateSpaceHelpers-impl.hpp

```diff
@@ -56,7 +56,11 @@ inline Eigen::VectorXd getPositionUpperLimits(
 
 //=============================================================================
 using JointStateSpaceTypeList = util::type_list<
-  statespace::dart::RnJoint,
+  statespace::dart::R0Joint,
+  statespace::dart::R1Joint,
+  statespace::dart::R2Joint,
+  statespace::dart::R3Joint,
+  statespace::dart::R6Joint,
```

Comment:
All the `RnX` classes support `Eigen::Dynamic`, but not for `RnJoint`. This is because all the dimensions of joints are determinant in compile time. More importantly, dynamic size dimension is not supported by none of DART's joint classes.